badparking/mobile_api/views.py

```diff
@@ -1,3 +1,40 @@
-from django.shortcuts import render
+from django.http import Http404
+from django.contrib.auth.models import User
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
 
-# Create your views here.
+from core.models import Claim, CrimeType
+from media.models import MediaFileModel
+from .serializers import ClaimSerializer
+
+
+class ClaimList(APIView):
+    def get(self, request, format=None):
+        claims = Claim.objects.all()
+        serializer = ClaimSerializer(claims, many=True)
+        return Response(serializer.data)
+
+    def post(self, request, format=None):
+        serializer = ClaimSerializer(data=request.data)
+
+        if serializer.is_valid():
+            claim = Claim(**serializer.data)
+            # temporary solution until we'll have auth
+            claim.user = request.user if request.user.id else User.objects.all()[0]
+            claim.save()
+
+            if (request.FILES):
+                for img in request.FILES.getlist('images'):
+                    image = MediaFileModel()
+                    image.file.save(img.name, img)
+                    image.save()
+                    claim.images.add(image)
+
+            types = request.data.getlist('types')
+            if (types):
+                ct = CrimeType.objects.filter(name__in = types)
```

Comment:
Also it's quite odd to accept names instead of ids... Names can change. The client will retrieve the types from the API anyway as I understand it. Even if not, it's better to use some slugs or whatever that doesn't change with the name.